sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -565,7 +579,62 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
           children)
       val (db, tableName) = extractDbNameTableName(tableNameParts)
 
-      CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, Some(node))
+      var tableDesc =
+        HiveTable(
+          specifiedDatabase = db,
+          name = tableName,
+          schema = Seq.empty,
+          partitionColumns = Seq.empty,
+          properties = Map.empty,
+          serdeProperties = Map.empty,
+          tableType = ManagedTable,
+          location = None,
+          inputFormat = None,
+          outputFormat = None,
+          serde = None)
+
+      // TODO: Handle all the cases here...
```

Comment:
@chenghao-intel, do you have time to help finish the implementation here.  I had to remove your version that relied on Hive's semantic analyzer.